src/opensearch_sdk_py/settings/settings.py

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+
+
+from typing import Any, Dict, Optional, Union
+
+from opensearch_sdk_py.transport.stream_input import StreamInput
+from opensearch_sdk_py.transport.stream_output import StreamOutput
+
+
+class Settings:
```

Comment:
I would inherit from `Dict`. Here's an implementation that does: https://github.com/opensearch-project/opensearch-build/blob/dd7e905cfd9b34562040ba9d78e221960dbeda11/src/manifests/manifests.py#L19

Reply:
> I would inherit from `Dict`.

I considered that but
 - there are actually two internal map-like structures.  There's a `secure_settings` (not yet implemented) which acts similarly to a string-to-string map (including a `get()` method) and it's reasonable to keep them on the same level for consistency
 - we use a `get()` method which would conflict with the `Dict` `get()` method (or at least return unexpected results)

Reply:
For the first one I wonder whether "secure setting" should be expressed as "a kind of setting". 

In what way are we reimplementing `get`? I would expect that populating settings changes, but retrieval should be a direct lookup, no?

Reply:
> For the first one I wonder whether "secure setting" should be expressed as "a kind of setting".

It's actually an interface with multiple implementations. Here's one: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/KeyStoreWrapper.java

I'd not want to stray too far from the existing implementation.

> In what way are we reimplementing `get`? 

Our version returns a `String` (`str`) not an object.  So an integer or boolean returns its string representation; an array/list object returns a string with square brackets; a map returns a key-value map with curly braces (basically JSON), a time value will include time units, byte sizes will include byte units, etc.  
https://github.com/opensearch-project/opensearch-sdk-py/blob/ef9a6b019b4a1e8134f02ced8cc79d6f5fdce7eb/tests/settings/test_settings.py#L39-L46

These Strings will be parsed by our future implementation of the `Setting<T>` class, each type having its own parser for the `str`.